### DISCUSSION TITLE
Unable to run a prebuild script

### DISCUSSION THREAD

**Discussion by claudia-meneghesso** [vercel/next.js#37115] [Help] [answered]
Hello everyone!

I have been trying to apply this [method](https://mmazzarolo.com/blog/2021-04-27-nextjs-robots-txt/) for generating a robots.txt file on Nextjs. Literally as described in the blog post. However, I am getting the below build error:

```
Run yarn prebuild
yarn run v1.22.18
$ scripts/generate-robots-txt
/bin/sh: 1: scripts/generate-robots-txt: not found
error Command failed with exit code 12[7](https://github.com/...../runs/6551722937?check_suite_focus=true#step:5:8).
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 127.
```

Seems like the prebuild script is not finding the file. I have tried to play around the path a bit (adding the file extension, adding a slash at the start of the path, etc) but this is still not working. Any suggestions?

**Comment by icyJoseph** [comment, marked as answer]
It is a bit strange, if you think about it, why would `generateRobotsTxt` work at all without being invoked. Somehow, somewhere, someone has to invoke the function. I know that you can make shell scripts but that requires a few steps not listed there.

I think the author might have made a small mistake, try like this:

```js
const fs = require("fs");

const crawlableRobotsTxt = `User-agent: *\nAllow: /`;

const uncrawlableRobotsTxt = `User-agent: *\nDisallow: /`;

function generateRobotsTxt() {
  // Create a non-crawlable robots.txt in non-production environments
  const robotsTxt =
    process.env.VERCEL_ENV === "production"
      ? crawlableRobotsTxt
      : uncrawlableRobotsTxt;

  // Create robots.txt file
  fs.writeFileSync("public/robots.txt", robotsTxt);

  console.log(
    `Generated a ${
      process.env.VERCEL_ENV === "production" ? "crawlable" : "non-crawlable"
    } public/robots.txt`
  );
}

generateRobotsTxt();
```

And then:

```js
  "prebuild": "node scripts/generate-robots-txt",
```

Maybe this is supposed to work on yarn 2? I don't know.  

If you want to use it in the `next.config.js` file instead, then do `module.exports = generateRobotsTxt` and invoke it there in the webpack property.

**Reply by claudia-meneghesso** [reply]
Thanks [icyJoseph](https://github.com/icyJoseph)! this approach worked out fine 😊  Cheers!
